sdk/identity/identity/src/client/identityClient.ts

```diff
@@ -192,16 +216,63 @@ export class IdentityClient extends ServiceClient implements INetworkModule {
     }
   }
 
+  // Here is a custom layer that allows us to abort requests that go through MSAL,
+  // since MSAL doesn't allow us to pass options all the way through.
+
+  generateAbortSignal(correlationId?: string): AbortSignalLike {
+    const controller = new AbortController();
+    const key = correlationId || noCorrelationId;
+
+    const controllers = this.abortControllers.get(key) || [];
+    controllers.push(controller);
+    this.abortControllers.set(key, controllers);
```

Comment:
So these get cleaned up when `abortRequests` is called, but what if abort is never called? We're not holding onto them forever, are we?

Reply:
I’ll address this in v2: https://github.com/Azure/azure-sdk-for-js/issues/16436